### PR TITLE
Properly search for the colon when checking for type hints in function arguments

### LINKED ISSUE #158: Fix multi-token type hints in line formatter
The line formatter handles the spacing of `=` inside of a function declaration with a type hint by looking two tokens behind the `=` to see if there's a colon. This works for simple declarations like `def foo(a: str = "bar")`, but does not work for more complicated types such as `def foo(a: Union[int, str] = "bar")`.

A more correct method would be to look for a colon anywhere behind the `=` until reaching a `(` or `,`.

### PR DESCRIPTION
**Pull request by jakebailey**
Fixes #158.

### CODE REVIEW

File: src/LanguageServer/Impl/Implementation/LineFormatter.cs [hunk 1]
```diff
@@ -146,7 +146,31 @@ public class LineFormatter {
...
+                                }
+
+                                if (p.Kind == TokenKind.Comma) {
+                                    // Found a comma, indicating the end of another argument, so stop.
+                                    break;
+                                }
+
+                                if (p.Kind == TokenKind.Colon) {
+                                    // Found a colon before hitting another argument or the opening parenthesis, so add spacing.
+                                    goto case TokenKind.AddEqual;
```
Comment (AlexanderSher) [inline src/LanguageServer/Impl/Implementation/LineFormatter.cs]: Please don't use `goto`.
I'd actually extract the whole case body into separate method and use `case … when …`

Comment (AlexanderSher) [reply src/LanguageServer/Impl/Implementation/LineFormatter.cs]: > against using goto 

I'm actually against fall-throughs from one non-empty case to another. Leave it as it is for now, I'll take a look myself.

Comment (MikhailArkhipov) [reply src/LanguageServer/Impl/Implementation/LineFormatter.cs]: It can bet something like HandleABC() function called from two cases.

**Comment by AlexanderSher** [comment]
Closed in favor of #168.

**Pull request closed without merging**